third_party/move/tools/move-mutator/src/mutate.rs

```diff
@@ -54,25 +57,36 @@ fn traverse_module_with_check(
     files: &FilesSourceText,
 ) -> anyhow::Result<Vec<Mutant>> {
     // We need to check if module comes from our source tree or from the deps, as we don't want to traverse
-    // all the dependencies.
+    // all the dependencies. That's a bit tricky as global deps are easy to identify but local deps can be
+    // anywhere near the project tree.
     let (ident, _) = &module;
     let (filename, _) = files.get(&ident.loc.file_hash()).unwrap(); // File must exist inside the hashmap so it's safe to unwrap.
     let filename_path = Path::new(filename.as_str());
 
-    let mut paths_to_check = conf.project.move_sources.clone();
-    if let Some(project_path) = &conf.project_path {
-        paths_to_check.push(project_path.clone());
-    }
-
-    if !paths_to_check
-        .iter()
-        .any(|path| filename_path.starts_with(path))
+    if !conf.project.move_sources.is_empty()
+        && !conf
+            .project
+            .move_sources
+            .contains(&filename_path.to_path_buf())
     {
         trace!(
             "Skipping module {} as it does not come from source project",
             filename
         );
         return Ok(vec![]);
+    } else if conf.project.move_sources.is_empty() {
```

Comment:
nit: unconditional return from the above `if` block so `else` is not needed here
```suggestion
    }
    
    if conf.project.move_sources.is_empty() {
```

Reply:
Removed :+1: